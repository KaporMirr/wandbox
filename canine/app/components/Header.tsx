import React, { useState } from "react";
import Navbar from "react-bootstrap/Navbar";
import { Modal, Nav, NavDropdown } from "react-bootstrap";
import { EditorSettings } from "./Editor/EditorSettings";
import { useEditorContext } from "~/contexts/EditorContext";
import { useSidebarContext } from "~/contexts/SidebarContext";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface HeaderProps {}

const Header: React.FC<HeaderProps> = (): React.ReactElement => {
  const [showSettings, setShowSettings] = useState<boolean>(false);
  const editor = useEditorContext();
  const sidebar = useSidebarContext();
  const { githubUser } = WANDBOX_LOADER_DATA;

  return (
    <Navbar
      className="px-16px"
      bg="primary"
      variant="dark"
      style={{ height: 48 }}
    >
      <Navbar.Brand href="/">Wandbox</Navbar.Brand>
      <Navbar.Toggle />
      <Navbar.Collapse className="justify-content-end">
        <Nav>
          <NavDropdown title="Tools" align="end">
            <NavDropdown.Item onClick={() => setShowSettings(true)}>
              Editor Settings
            </NavDropdown.Item>
            <NavDropdown.Item
              onClick={() => {
                if (sidebar.opened) {
                  sidebar.setOpened(false);
                  sidebar.setLocked(false);
                } else {
                  sidebar.setOpened(true);
                }
              }}
            >
              History
            </NavDropdown.Item>
          </NavDropdown>
          <Nav.Link
            target="_blank"
            rel="noopener noreferrer"
            href="https://github.com/melpon/wandbox"
          >
            GitHub
          </Nav.Link>
          {githubUser === null ? (
            <Nav.Link
              href={`https://github.com/login/oauth/authorize?client_id=${WANDBOX_GITHUB_CLIENT_ID}`}
            >
              Login
            </Nav.Link>
          ) : (
            <NavDropdown
              title={
                <img
                  src={githubUser.avatar_url}
                  style={{ width: 24, height: 24, borderRadius: 12 }}
                />
              }
              align="end"
            >
              <NavDropdown.Item href="/logout">Logout</NavDropdown.Item>
            </NavDropdown>
          )}
        </Nav>
      </Navbar.Collapse>
      <Modal show={showSettings} onHide={() => setShowSettings(false)}>
        <Modal.Body className="d-flex flex-column">
          <button
            type="button"
            className="align-self-end btn-close"
            data-bs-dismiss="modal"
            aria-label="Close"
            onClick={() => setShowSettings(false)}
          />
          <EditorSettings settings={editor.settings} />
        </Modal.Body>
      </Modal>
    </Navbar>
  );
};

export { Header };
