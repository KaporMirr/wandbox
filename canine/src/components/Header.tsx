import React from "react";
import Link from 'next/link'
import Navbar from "react-bootstrap/Navbar";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface HeaderProps {}

const Header: React.FC<HeaderProps> = (): React.ReactElement => {
  return (
    <Navbar>
      <Navbar.Brand as={Link} href="/">
        Wandbox
      </Navbar.Brand>
    </Navbar>
  );
};

export { Header };
