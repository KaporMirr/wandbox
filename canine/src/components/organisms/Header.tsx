import React from "react";
import { Link } from "react-router-dom";
import makeStyles from "@material-ui/styles/makeStyles";
import { Theme } from "@material-ui/core/styles/createMuiTheme";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import deepOrange from "@material-ui/core/colors/deepOrange";

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type, @typescript-eslint/no-unused-vars
const useStyles = makeStyles((theme: Theme) => ({
  root: {
    flexGrow: 1,
  },
  appBar: {},
  brand: {
    textDecoration: "none",
    flexGrow: 1,
  },
  avatar: {
    color: "#fff",
    backgroundColor: deepOrange[500],
  },
}));

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface HeaderProps {}

const Header: React.FC<HeaderProps> = (): React.ReactElement => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static" className={classes.appBar}>
        <Toolbar>
          <Typography
            className={classes.brand}
            variant="h6"
            color="inherit"
            // component={Link} to="/" だと型が合わなくて怒られる。そのうち直るかもしれない。
            // https://github.com/mui-org/material-ui/issues/8598#issuecomment-399184959
            // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
            component={(props) => <Link to="/" {...props} />}
          >
            Wandbox
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export { Header };
