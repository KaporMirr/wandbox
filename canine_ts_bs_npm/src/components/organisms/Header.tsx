import React from "react";
import { Link } from "react-router-dom";
import Navbar from "react-bootstrap/Navbar";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface HeaderProps {}

const Header: React.FC<HeaderProps> = (): React.ReactElement => {
  return (
    <Navbar>
      <Navbar.Brand as={Link} to="/">
        Wandbox
      </Navbar.Brand>
    </Navbar>
  );
};

export { Header };
