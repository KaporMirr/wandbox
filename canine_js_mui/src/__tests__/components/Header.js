import React from 'react'
import Header from 'components/Header'
import renderer from 'react-test-renderer'

test('ヘッダーのスナップショットが意図せず変わっていない', () => {
  const component = renderer.create(<Header />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
