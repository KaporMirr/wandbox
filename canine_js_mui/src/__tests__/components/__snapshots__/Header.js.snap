// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ヘッダーのスナップショットが意図せず変わっていない 1`] = `
<div>
  <header
    className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-5 MuiAppBar-positionAbsolute-7 MuiAppBar-colorPrimary-11 Header-appBar-1"
  >
    <div
      className="MuiToolbar-root-40 MuiToolbar-gutters-41"
    >
      <a
        className="MuiTypography-root-42 MuiTypography-title-48 MuiTypography-colorInherit-61 Header-brand-2"
        href="/"
        onClick={[Function]}
      >
        gPaaS
      </a>
    </div>
  </header>
  <div
    className="MuiDrawer-docked-66"
  >
    <div
      className="MuiPaper-root-13 MuiPaper-elevation0-15 MuiDrawer-paper-67 Header-drawerPaper-3 MuiDrawer-paperAnchorLeft-68 MuiDrawer-paperAnchorDockedLeft-72"
    >
      <div
        className="Header-toolbar-4"
      />
      <ul
        className="MuiList-root-77 MuiList-padding-78"
      >
        <a
          className="MuiButtonBase-root-91 MuiListItem-root-81 MuiListItem-default-84 MuiListItem-gutters-88 MuiListItem-button-89"
          href="/project/sysop"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-94"
          >
            <h3
              className="MuiTypography-root-42 MuiTypography-subheading-49 MuiListItemText-primary-97"
            >
              sysop
            </h3>
          </div>
          <span
            className="MuiTouchRipple-root-100"
          />
        </a>
        <a
          className="MuiButtonBase-root-91 MuiListItem-root-81 MuiListItem-default-84 MuiListItem-gutters-88 MuiListItem-button-89"
          href="/project/gscxbase"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <div
            className="MuiListItemText-root-94"
          >
            <h3
              className="MuiTypography-root-42 MuiTypography-subheading-49 MuiListItemText-primary-97"
            >
              gscxbase
            </h3>
          </div>
          <span
            className="MuiTouchRipple-root-100"
          />
        </a>
      </ul>
    </div>
  </div>
</div>
`;
